Enable default component arguments (#199)
ml6team/fondant: docs/component_spec.md

### Args

The `args` section describes which arguments the component takes. Each argument is defined by a 

> `description` and a `type`, which should be one of the builtin Python types. Additionally, you can 
> pass an optional argument in the optional `default` field of the arguments. This can be useful in case
> your function has many arguments that could be set to default and that don't need to be explicitly defined when
> initializing your component. 

Reviewer: ```suggestion
`description` and a `type`, which should be one of the builtin Python types. Additionally, you can 
set an optional `default` value for each argument.
```

_Note:_ default iterable arguments such as `dict` and `list` have to be passed as a string 
(e.g. `'{"foo":1, "bar":2}`, `'["foo","bar]'`)
```yaml
args:
  custom_argument:
    description: A custom argument
    type: str
  default_argument:
    description: A default argument
    type: str
    default: bar
``` 

> These arguments are passed in when the component is instantiated. Notice that we are not passing the
> default argument specified above. You could override the default value in the component spec by passing
> it as an argument to the component.

Reviewer: ```suggestion
These arguments are passed in when the component is instantiated. 
If an argument is not explicitly provided, the default value will be used instead if available.```
